### ISSUE TITLE
Variables not passed as an array with a GET request

### ISSUE THREAD

**Issue by askogrand** [markhuot/craftql#71]
Based on the conversation here:
https://github.com/markhuot/craftql/issues/12#issuecomment-326855140

it was pretty easy to start using GET vs POST. It worked for all queries that didnt contain variables, but when I did pass variables, I received this error:

```Argument 1 passed to GraphQL\Validator\Rules\QueryComplexity::setRawVariableValues() must be of the type array or null, string given, called in /home/vagrant/thirdway/vendor/webonyx/graphql-php/src/GraphQL.php on line 137```

Looking into the api controller, I see that we dont actually decode the get variable from the request, and so a string is passed on as the variable in the graphql request. 

https://github.com/markhuot/craftql/blob/master/src/Controllers/ApiController.php#L102

Changing line 102 to `$variables = json_decode($query, true);` solves this issue, although there may be use cases I cant foresee.

**Comment by markhuot** [comment]
So, originally, I expected GET parameters to be passed as actual parameters, e.g.,

```
/api?query=query GetEntries($limit:Int){entries(limit:$limit){id}}&variables[limit]=1
```

And that works, for the most part. Unfortunately, URL encoded params can't distinguish between string and integer types so the above query yells as you because it expects limit to be a number but PHP makes all query params strings.

So, `json_decode` does seem to be the way to go. This is pushed to `dev-master` if you want to update your `composer.json` and take advantage of it today.

**Comment by markhuot** [comment]
This was released with 1.0.0
